src/masternode.h

```diff
@@ -166,6 +166,37 @@ class CMasternode
         MASTERNODE_POSE_BAN
     };
 
+    enum CollateralStatus {
+        COLLATERAL_OK,
+        COLLATERAL_UTXO_NOT_FOUND,
+        COLLATERAL_INVALID
+    };
+
+    static CollateralStatus CheckCollateral(CTxIn vin)
```

Comment:
Move implementation to cpp and definition way down, somewhere below `swap` would be ok